pkg/scheduler/schedule_one_test.go

```diff
@@ -3404,6 +3411,112 @@ func TestPreferNominatedNodeFilterCallCounts(t *testing.T) {
+			pod := st.MakePod().Name(fmt.Sprintf("pod-%d", i)).
+				UID(fmt.Sprintf("pod-%d", i)).
+				NodeAffinityIn(metav1.ObjectNameField, []string{nodeName}).
+				Obj()
+			// bind pod with each node
+			pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchFields = []v1.NodeSelectorRequirement{
+				{
+					Key:      metav1.ObjectNameField,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{nodeName},
+				},
+			}
+			pod.Spec.SchedulerName = "default-scheduler"
+			pods = append(pods, pod)
+		}
+		nodes = makeNodeList(nodeNameList)
+		nodeObjs := make([]runtime.Object, 0, nodeNum)
+		for i := 0; i < nodeNum; i++ {
+			nodeObjs = append(nodeObjs, nodes[i])
+		}
+
+		client := clientsetfake.NewSimpleClientset(nodeObjs...)
+		informerFactory := informers.NewSharedInformerFactory(client, 0)
+		pluginRegistrations := []tf.RegisterPluginFunc{
+			tf.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+			tf.RegisterPreFilterPlugin(nodeaffinity.Name, nodeaffinity.New),
+			tf.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+		}
+		broadcaster := events.NewBroadcaster(&events.EventSinkImpl{Interface: client.EventsV1()})
+		registry := frameworkruntime.Registry{}
+		profiles := schedulerapi.KubeSchedulerProfile{
+			SchedulerName: "default-scheduler",
+			Plugins:       &schedulerapi.Plugins{},
+		}
+		for _, fn := range pluginRegistrations {
+			fn(&registry, &profiles)
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		sched, err := New(ctx, client, informerFactory, nil, profile.NewRecorderFactory(broadcaster),
+			WithProfiles(profiles))
+		if err != nil {
+			b.Fatal(err)
+		}
+
+		wg := sync.WaitGroup{}
+		wg.Add(nodeNum)
+		bindings := make(map[string]string)
+		client.PrependReactor("create", "pods", func(action clienttesting.Action) (bool, runtime.Object, error) {
+			if action.GetSubresource() != "binding" {
+				return false, nil, nil
+			}
+			binding := action.(clienttesting.CreateAction).GetObject().(*v1.Binding)
+			bindings[binding.Name] = binding.Target.Name
+			wg.Done()
+			return true, binding, nil
+		})
+		if err != nil {
+			b.Fatal(err)
+		}
+
+		// Run scheduler.
+		informerFactory.Start(ctx.Done())
+		informerFactory.WaitForCacheSync(ctx.Done())
+		if err = sched.WaitForHandlersSync(ctx); err != nil {
+			b.Fatalf("Handlers failed to sync: %v: ", err)
+		}
+		go sched.Run(ctx)
+
+		start := time.Now()
+		logger := klog.FromContext(ctx)
+		// Send pods to be scheduled.
+		for _, p := range pods {
+			sched.SchedulingQueue.Add(logger, p)
+		}
+		wg.Wait()
+		cancel()
+		cost := time.Since(start).Seconds()
+		throughput := float64(nodeNum) / cost
+		if throughput < benchmarkThroughput {
+			b.Fatalf("ScheduleOne_onePodPerNode: %d nodes, %d pods, cost: %f seconds, throughput: %f pods/second, below the baseline throughput standard: %f pods/second",
+				nodeNum, nodeNum, cost, throughput, benchmarkThroughput)
+		}
```

Comment:
@sanposhiho Is it possible to add `benchmarkThroughput` to this benchmark test, at least to verify that the code quality is not lower than a certain standard?

Of course, the throughput in the actual production environment will be affected by more factors such as network memory. I wonder if we can use [kwok](https://github.com/kubernetes-sigs/kwok) in e2e to help us better simulate the actual environment. The current benchmark test focuses more on code quality.

Reply:
At least in this benchmark test, we can see that preallocation for NodeToStatusMap helps improve throughput.

Reply:
I'm not sure why that needs to be this type of benchmark test. scheduler_perf is a basic tool we usually use to measure performance. It also has a monitoring visualization provided via [perf-dash](https://perf-dash.k8s.io/). I believe we should create this kind of benchmark tests **only** when it's somehow not possible to measure the performance within scheduler_perf. 
(Besides that, either way, IIUC, Benchmark_XXX tests aren't run in prow jobs.)

But, I like your idea of keeping a certain throughput as tests. But, for that idea too, I'd like to see something like:
- we define a desired throughput for each test case of scheduler_perf.
- implement something so that we easily notice the degradation based on ^. e.g., the CI fails in PRs, we get alerts in Slack if the scheduler of the master branch is slow, etc.

Reply:
Starting a thread -
https://kubernetes.slack.com/archives/C09QZTRH7/p1715262959575039

Reply:
https://github.com/kubernetes/kubernetes/issues/124774